data-prepper-plugins/otel-trace-group-prepper/src/test/java/com/amazon/dataprepper/plugins/prepper/oteltracegroup/OtelTraceGroupPrepperTests.java

```diff
@@ -0,0 +1,256 @@
+            setPipelineName("testPipelineName");
+        }};
+        otelTraceGroupPrepper = new OtelTraceGroupPrepper(testPluginSetting);
+        executorService = Executors.newFixedThreadPool(TEST_NUM_WORKERS);
+    }
+
+    @After
+    public void tearDown() {
+        otelTraceGroupPrepper.shutdown();
+        connectionConfigurationMockedStatic.close();
+        executorService.shutdown();
+    }
+
+    @Test
+    public void testShutDown() throws IOException {
+        // When
+        otelTraceGroupPrepper.shutdown();
+
+        // Then
+        verify(restHighLevelClient, times(1)).close();
+    }
+
+    @Test
+    public void testTraceGroupFillSuccess() throws IOException {
+        // Given
+        Record<String> testRecord = buildRawSpanRecord(TEST_RAW_SPAN_MISSING_TRACE_GROUP_JSON_FILE_1);
+        List<Record<String>> testRecords = Collections.singletonList(testRecord);
+
+        // When
+        List<Record<String>> recordsOut = (List<Record<String>>) otelTraceGroupPrepper.doExecute(testRecords);
+
+        // Then
+        assertEquals(1, recordsOut.size());
+        Record<String> recordOut = recordsOut.get(0);
+        assertEquals(TEST_TRACE_GROUP_1, extractTraceGroupFromRecord(recordOut));
+    }
+
+    @Test
+    public void testTraceGroupFillFailDueToRequest() throws IOException {
+        // Given
+        Record<String> testRecord = buildRawSpanRecord(TEST_RAW_SPAN_MISSING_TRACE_GROUP_JSON_FILE_1);
+        List<Record<String>> testRecords = Collections.singletonList(testRecord);
+        when(restHighLevelClient.search(any(SearchRequest.class), any(RequestOptions.class)))
+                .thenThrow(new ElasticsearchException("Failure due to search request"));
+
+        // When
+        List<Record<String>> recordsOut = (List<Record<String>>) otelTraceGroupPrepper.doExecute(testRecords);
+
+        // Then
+        assertEquals(1, recordsOut.size());
+        Record<String> recordOut = recordsOut.get(0);
+        assertEquals(testRecord, recordOut);
+    }
+
+    @Test
+    public void testTraceGroupFillFailDueToNoHits() throws IOException {
+        // Given
+        Record<String> testRecord = buildRawSpanRecord(TEST_RAW_SPAN_MISSING_TRACE_GROUP_JSON_FILE_1);
+        List<Record<String>> testRecords = Collections.singletonList(testRecord);
+        when(restHighLevelClient.search(any(SearchRequest.class), any(RequestOptions.class))).thenReturn(testSearchResponse);
+        when(testSearchResponse.getHits()).thenReturn(testSearchHits);
+        when(testSearchHits.getHits()).thenReturn(new SearchHit[] {});
+
+        // When
+        List<Record<String>> recordsOut = (List<Record<String>>) otelTraceGroupPrepper.doExecute(testRecords);
+
+        // Then
+        assertEquals(1, recordsOut.size());
+        Record<String> recordOut = recordsOut.get(0);
+        assertEquals(testRecord, recordOut);
+    }
+
+    @Test
+    public void testTraceGroupFillFailDueToException() throws IOException {
```

Comment:
How is this different from `testTraceGroupFillFailDueToRequest()`?

Reply:
Oops. it is duplicate. Must have lost my memory while working.